lesson18/src/js/main.js

```diff
@@ -0,0 +1,299 @@
+  pagenationBullet.forEach((targetButton) => {
+    targetButton.addEventListener('click',(e) => {
+      const activePagenationBullet = document.querySelector('[aria-current="true"]');
+      const clickedPagenationBulletNo = e.currentTarget.getAttribute('data-pagenation-index');
+      activePagenationBullet.setAttribute('aria-current','false');
+      e.currentTarget.setAttribute('aria-current','true');
+      document.getElementById('js-current').textContent = clickedPagenationBulletNo;
+
+      const activeSlide = document.getElementsByClassName('slider__item is-active');
+      activeSlide[0].classList.remove('is-active');
+      document.querySelector(`[data-slide-index="${clickedPagenationBulletNo}"]`).classList.add('is-active');
+
+      const pagenationCurrent = document.getElementById("js-current");
+      pagenationCurrent.textContent = clickedPagenationBulletNo;
+      toggleButtonDisabled();
+      resetAutoPlaySlide();
+    })
+  })
+
+}
+
+function toggleButtonDisabled() {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const activeImage = document.querySelector(".is-active");
+  const currentIndex = Number(activeImage.dataset.slideIndex);
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+  prevButton.disabled = currentIndex === firstIndex;
+  nextButton.disabled = currentIndex === lastIndex;
+}
+
+function renderSlidContents(slideImageData){
+  renderSlideArea();
+  renderSlideImage(slideImageData);
+  renderPagenation(slideImageData.length);
+  renderPagenationBullet(slideImageData.length);
+  renderSlideButton();
+  addSwitchButtonEvent();
+  clickPagenationBullet();
+  toggleButtonDisabled();
+  autoPlayslide();
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(
+        fetchImageData(
+          //https://api.json-generator.com/templates/IZjWl012CAMD/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl : image = 0
+          //https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&status=503 : 503
+          "https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl"
+        )
+      );
+    }, 3000);
+  });
+}
+
+async function fetchImageData(URL) {
+  const response = await fetch(URL);
+  if (response.ok) {
+    const json = await response.json();
+    return json;
+  } else {
+    console.error(`${response.status}:${response.statusText}`);
+    displayErrorMassage("サーバの通信に失敗しました");
+  }
+}
+
+async function getSlideImage(){
+  showLoadingImage();
+  try{
+     const json = await callImageData();
+     const slideData = json.slide;
+    if(slideData.length > 0){
+      renderSlidContents(slideData);
+    }
+  }catch{
+    addNoimage();
```

Comment:
Q.
It says `addNoimage`, is that the image to run if there is no image to come across?

Currently, the catch clause is passed only when the fetch fails, so I don't think it will be executed even if the image is 0.

Reply:
Thank you for your review!

Certainly the image is 0
Since the addNoimage () function is not executed at this time, I added a condition.

[4b7c353](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/4b7c353b8a227dbc2b2f84dcf72eb1473c5b9acd)